spring-integration-test/src/main/java/org/springframework/integration/test/context/MockIntegrationContext.java

```diff
@@ -112,15 +120,31 @@ public void instead(String pollingAdapterId, MessageSource<?> mockMessageSource,
 		instead(pollingAdapterId, mockMessageSource, SourcePollingChannelAdapter.class, "source", autoStartup);
 	}
 
-	private void instead(String endpointId, Object mock, Class<?> endpointClass, String property,
+	public void instead(String consumerEndpointId, MessageHandler mockMessageHandler) {
+		instead(consumerEndpointId, mockMessageHandler, true);
+	}
+
+	public void instead(String consumerEndpointId, MessageHandler mockMessageHandler, boolean autoStartup) {
+		instead(consumerEndpointId, mockMessageHandler, IntegrationConsumer.class, "handler", autoStartup);
+		if (mockMessageHandler instanceof MessageProducer) {
```

Comment:
Well, you always can `stop()` any Inbound Gateways or `MessageProducer`s.
Remember `@SpringIntegrationTest(noAutoStartup)`?
And provide an `@InboundChannelAdapter` for the same `requestChannel` and use `MockIntegration.mockMessageSource()` to produce messages for testing.

So, I don't feel like we should right now spend time thinking how useful to mock Inbound Gateways.